Jenkinsfile

```diff
@@ -39,13 +56,38 @@ pipeline {
                 sh './mvnw clean install'
             }
         }
-    }
 
+        /*
+        TODO uncomment and finish up the command here once we're ready to release snapshots
+        stage('Publish snapshot') {
+            when {
+                branch 'master'
+            }
+            steps {
+                sh './mvnw $TODO'
+            }
+        }
+         */
+    }
 
     post {
         always {
             junit '**/target/surefire-reports/*.xml'
             deleteDir()
         }
+
+        changed {
+            script {
+                if (env.BRANCH_NAME == 'master') {
```

Comment:
Does this or can we distinguish between `master` on forks (e.g. some pull requests might use the master branch in their fork)?

Reply:
Great point. I *think* those wouldn't trigger this build, but better be on the safe side, I'll add an extra check.

Reply:
Alright, I've opened a PR to verify - #20. It seems we're fine:

* Commits on my fork didn't trigger the build (as expected)
* On the PR build the `BRANCH_NAME` env var is `PR-20`; two other env vars `CHANGE_BRANCH` and `TARGET_BRANCH` describe the actual branches involved (see https://builds.apache.org/blue/organizations/jenkins/incubator-zipkin-brave-karaf/detail/PR-20/1/pipeline/37)